doc/specifications/proposed/content_download/content_download.md

```diff
@@ -0,0 +1,60 @@
+# Content download
+
+> Author: bd@ez.no
+
+## Summary
+This feature covers downloading of Content binary files over HTTP.
+
+It covers:
+- generation of proper links from Twig as well as PHP,
+- permissions checking (the user must be allowed to read the Content)
+- streaming of the file over HTTP
+
+## Components
+
+### Twig helper
+
+> Name: ez_download_path
+> Usage: `ez_download_path( content, 'file' )`
+
+The helper will be used to generate a download link for a given Field/Content.
+
+### Controller action + route
+
+#### Route
+
+Path: `/content/download/{contentId}/{fieldIdentifier}/{fileName}`
+
+Example: ``/content/download/68/file/My file.pdf
+
+Query parameters
+
+- version (optional)
+
+  The version number the file must be downloaded for. Requires the versionview permission.
+  If not specified, the published version is used.
```

Comment:
In order to cache without requiring purge, yes, we'd have to add it to the URI, or it might not be updated.

One drawback of having it in, though, is that when a new version is published, all existing (saved, cached) links to the file become obsolete, and result in a Forbidden error (requires versionview).